File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        # initialize dict
        seen = {}                                   

        # iterate over index and num in array nums
        for i, num in enumerate(nums):              
            diff = target - num
            # if complement of current num exists in dict, return indices of complement and current num
            if diff in seen:                    
                return [seen[diff], i]
            # append current num and its array index to dict          
            seen[num] = i                      

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> map = new HashMap<>();            // initialize new hashmap

        for (int i = 0; i < nums.length; i++) {                     // iterate over elements in nums
            if (map.containsKey(target - nums[i])) {                // if map contains difference between target and element
                return new int[] {map.get(target - nums[i]), i};    // return indices of element and difference
            }
            map.put(nums[i], i);                                    // else update hashmap with element, index
        }
        return new int[] {};                                      
    }
}